### DISCUSSION TITLE
3D Flow around object example

### DISCUSSION THREAD

**Discussion by BimwerxNZ** [su2code/SU2#2262] [Q&A] [answered]
Would it be possible to add a 3D example of RANS, SST model with wind tunnel (6 walls) and an obstacle to simulate wind flow?

Ultimately I'd like to see how to retrieve resulting pressures on a surface from the model, after providing inlet flow.

I've tried adapting Multiphysics FSI 2D model with no luck, too many possible input issues on my end, and the SU2_CFD executable reports no errors when loading config, just finishes with no output or results.

Meshes I've tried were generated from GMSH (tets for wind tunnel volume and tri for obstacle as a planar surface).

I'm a new user so please excuse my lack of knowledge using SU2. An example would be great help! Thanks!

**Comment by bigfooted** [comment]
There is nothing preventing you from simulating the entire wind tunnel with obstacles in it. What is it that you really try to achieve, because you also mention FSI.

**Reply by BimwerxNZ** [reply]
I'm sure it is possible, the problem is that I don't know how using SU2. Ultimately I just want to determine resulting forces on walls of a building by defining the outer geometry, and a flow velocity or wind speed source. I've used cfdOF before, so I assume the principles would be similar (but not sure how to set it up correctly).

**Reply by BimwerxNZ** [reply]
I've tried the attached approach, but result in 'SU2 has diverged (NaN detected).' if I attempt to use no-slip walls for the interior members.
[Wind Simulation.zip](https://github.com/su2code/SU2/files/14969251/Wind.Simulation.zip)

**Reply by bigfooted** [reply]
First, is your flow compressible? If not, then look at some of the INC_RANS examples in the Testcases folder. The meshes are in the separate github Testcases repository.
I looked at your mesh and it is very coarse, and there is little room around your object. The inlet and outlet are also very close to the object so their presence will influence the simulation results around your object and might cause solver divergence. Try placing the inlets and outlets at least a distance of 10 "object size" away from the object. If you have turbulent flow, you need a much finer mesh close to the walls. Please have a look at some of the tutorials or the other testcases that we have in the repository. The y+ value has to be around 1 to be really accurate.

**Reply by BimwerxNZ** [reply]
Thank you for your response - I will try again with finer mesh and increased distance between inlet and outlet.

**Reply by bigfooted** [reply]
Look for instance at pictures of the CFD mesh around the Ahmed body or studies on the wind tunnel wall effect to see some examples. But does it need to be inside a wind tunnel? Is your goal to look at the effects of wind tunnel walls on the lift and drag of the object? If not, then you're better of simulating your object in a large sphere where the edge of the sphere is modeled as 'far field'.

**Reply by BimwerxNZ** [reply]
I'm only interested in determining wind load (pressure) on walls for buildings. With my limited knowledge of SU2 I assumed the way to do this is to create a virtual wind tunnel with inlet and outlet, and the obstacle in the center of it. I'd be interested in understanding how the sphere setup would work - could you perhaps be so kind as to point me to a relevant example? Thank you for the info!

**Reply by bigfooted** [reply, marked as answer]
For instance in this tutorial we do something like that: https://su2code.github.io/tutorials/Turbulent_ONERAM6/
You actually do not want to have the walls of the windtunnel in the way, so you increase the cell size as you move further away from the body of interest and far away from the body you place a farfield condition, which is basically the undisturbed flow velocity.

**Reply by BimwerxNZ** [reply]
Thank you so much, this really helped!
